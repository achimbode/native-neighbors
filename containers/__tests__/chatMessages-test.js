import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Messages from '../ChatMessages';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Messages', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Messages 
        username={dummyProps.username}
        color={dummyProps.color}
        users={dummyProps.users}
        messages={dummyProps.messages} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
}); 


const dummyProps = {
  username: 'Andrea',
  color: 'fuchsia',
  users: {

    'Andrea': {
      avatar_color: 'navy'
    }
    ,
    'Achim': {
      avatar_color: 'navy'
    },
  },
  messages: [
    {
      from: 'Andrea',
      msg: 'hello Achim',
    },
    {
      from: 'Achim',
      msg: 'hello Andrea',
    }
  ]
}