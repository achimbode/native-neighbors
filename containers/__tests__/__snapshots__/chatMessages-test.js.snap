// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages should render correctly 1`] = `
<View
  className="MessagesContainer"
>
  <View
    className="Messages"
  >
    <View
      key="0"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        className="userMsg"
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "fuchsia",
          }
        }
      >
        hello Achim
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Username
      </Text>
    </View>
    <View
      className="neighborsMsg"
      key="1"
      style={
        Object {
          "backgroundColor": "navy",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        hello Andrea
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Other Mesg
      </Text>
    </View>
  </View>
  <View
    className="IsTypingContainer"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      className="isTyping"
      ellipsizeMode="tail"
    >
      this.renderTyping()
    </Text>
  </View>
</View>
`;
